### ISSUE TITLE
Demandware language support (.isml & .ds)

### ISSUE THREAD

**Issue by AaronV** [Glavin001/atom-beautify#923]
## Description

Any chance of support for Demandware languages: .isml and .ds. These are very similar to .html and .js respectively.

**Comment by prettydiff** [comment]
Would you happen to have a link to the language spec?  It would help us evaluate if additional support is needed or if the language can be supported directly in existing beautifiers.

**Comment by AaronV** [comment]
Unfortunately, I've found relatively little information about the spec. The best I've seen is for ISML and is a list of their custom tags and this bit of description.

> Demandware ISML tags are extensions of the tag-based language conforming to SGML standards. They are commands inserted into a document that determine how Demandware eCommerce embeds live data into a page, and how to format this data together with the regular HTML code.

Is that enough to roll it into the HTML beautifier?

**Comment by prettydiff** [comment]
The currently supported HTML beautifiers are:

* http://jsbeautifier.org/
* http://prettydiff.com

Try running some code samples through those tools.  If the code samples aren't mutilated then we can presume support from current beautifiers.

**Comment by AaronV** [comment]
PrettyDiff seems to work fairly well for .isml and .ds files. Is it user-configurable to get those the html & js beautifiers working for those file-types? I ran into some cases where it didn't work well and wouldn't want to force those beautifiers on everyone using my workflow.

**Comment by prettydiff** [comment]
I recommend opening issues at one of those projects:

* https://github.com/prettydiff/prettydiff/issues
* https://github.com/beautify-web/js-beautify/issues

Please open an issue for any defect you notice with code samples.

**Comment by AaronV** [comment]
Thanks, Austin. Should this ticket be closed in the meantime?

**Comment by prettydiff** [comment]
No, leave this issue open and link to this issue from the beautifier specific issue.

**Comment by prettydiff** [comment]
It looks like Demandware was acquired by SalesForce in the months after this ticket was created.  Is Demandware still a thing?  If so is there a location for isml or ds documentation?

**Comment by zetxek** [comment]
It is - now it's called Salesforce Commerce Cloud. The syntax is still the same: `.isml` for template languages, `/ds` for legacy javascript (the new code is usually named `.js`)

**Comment by prettydiff** [comment]
@zetxek Do you know if there is a link to describes the `isml` spec?  I was on that site earlier today, but I could not find the documentation.

**Comment by zetxek** [comment]
@prettydiff the docs are behind an NDA - but I can share that it's an extension of the [SGML](https://searchmicroservices.techtarget.com/definition/SGML-Standard-Generalized-Markup-Language), with some specific additional tags that are valid for the syntax. Basically it's HTML + some proprietary tags (which can be both self closing or in the long format `<tag>whateverinside</tag>`).

I'm wondering if we could extract the syntax for an old Eclipse plugin that has syntax highlighting.

**Comment by prettydiff** [comment]
To make this work I just need to know which tag names are self closing, because otherwise they will be treated as start tags without an ending pair.  I can make this work in much the same way I extended support for ColdFusion's CFML.
